### PR TITLE
David/qblox exec params

### PR DESCRIPTION
**Pull request by DavidSarlle**
This Pr implements the use of the ExecutionParameters coded by @Jacfomg in qblox driver.

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.

### CODE REVIEW

File: src/qibolab/platforms/multiqubit.py [hunk 1]
```diff
@@ -45,7 +53,7 @@ def reload_settings(self):
         super().reload_settings()
         self.characterization = self.settings["characterization"]
         self.qubit_channel_map = self.settings["qubit_channel_map"]
-        self.nshots = self.settings["settings"]["nshots"]
+        self.nshots = self.settings["settings"]["hardware_avg"]
```
Comment (Jacfomg) [inline src/qibolab/platforms/multiqubit.py:56]: `hardware_avg` was agreed to be renamed to nshots so don't revert the changes

File: src/qibolab/platforms/multiqubit.py [hunk 2]
```diff
@@ -279,12 +287,22 @@ def disconnect(self):
...
+
+        if options.averaging_mode == AveragingMode.SINGLESHOT:
+            nshots = 1
+            self.average = False
+        elif options.averaging_mode == AveragingMode.CYCLIC:
+            nshots = options.nshots
+            self.average = True
+
+        if options.nshots is None:
             nshots = self.nshots
```
Comment (Jacfomg) [inline src/qibolab/platforms/multiqubit.py:302]: `options.nshots` will never be `None` at least on the new platform

File: src/qibolab/platforms/multiqubit.py [hunk 3]
```diff
@@ -332,16 +350,32 @@ def execute_pulse_sequence(self, sequence: PulseSequence, nshots=None):
...
+                        qres = acquisition_results[serial][1][0]
+                        if options.acquisition_type is AcquisitionType.RAW:
+                            if self.average:
+                                acquisition = AveragedRawWaveformResults(ires + 1j * qres)
+                            acquisition = RawWaveformResults(ires + 1j * qres)
+
+                        if options.acquisition_type is AcquisitionType.INTEGRATION:
+                            if self.average:
+                                acquisition = AveragedIntegratedResults(ires + 1j * qres)
+                            acquisition = IntegratedResults(ires + 1j * qres)
```
Comment (Jacfomg) [inline src/qibolab/platforms/multiqubit.py:372]: I would suggest to use `ExecutionParameters.results_type` as I mentioned

**Review by Jacfomg — CHANGES_REQUESTED** [review]

**Comment by Jacfomg** [comment]
@DavidSarlle if you are changing the things I commented it would be faster for me that when I recheck you resolve the conversations instead of leaving them there.

**Comment by DavidSarlle** [comment]
> @DavidSarlle if you are changing the things I commented it would be faster for me that when I recheck you resolve the conversations instead of leaving them there.

@Jacfomg I used to reply all the comments one by one as soon as the changes are ready and tested. So, Do not worry, you will be notified comment by comment.

File: src/qibolab/platforms/multiqubit.py (hunk 1 above)
Comment (DavidSarlle) [reply src/qibolab/platforms/multiqubit.py]: Sorry, I did not pushed the latest changes where the hardware_avg was changed to nshots. Now it should be ok.

File: src/qibolab/platforms/multiqubit.py (hunk 2 above)
Comment (DavidSarlle) [reply src/qibolab/platforms/multiqubit.py:302]: In the current qibocal main branch that is not true. For example:

imagen

The defaukt value of nshopts is None, so if nshots is not defined in the action runcard, the params.nshots could be None.

File: src/qibolab/platform.py [hunk 4]
```diff
@@ -77,7 +77,7 @@ def reload_settings(self):
             self.resonator_type = "3D" if self.nqubits == 1 else "2D"
 
         self.relaxation_time = settings["settings"]["relaxation_time"]
-        self.nshots = settings["settings"]["nshots"]
+        self.nshots = settings["settings"]["hardware_avg"]
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibolab/platform.py]: Sorry, I did not pushed the latest changes where the hardware_avg was changed to nshots. Now it should be ok.

File: src/qibolab/platforms/multiqubit.py (hunk 2 above)
Comment (Jacfomg) [reply src/qibolab/platforms/multiqubit.py:302]: That's not a problem. `ExecutionParameters` would get the nshots value defined on the runcard as default.

Comment (DavidSarlle) [reply src/qibolab/platforms/multiqubit.py]: it is not getting them... If I remove the init, the value is None.

Comment (Jacfomg) [reply src/qibolab/platforms/multiqubit.py]: From the platform

```
def execute_pulse_sequence(self, sequence, options, **kwargs):
        """Executes a pulse sequence.

        Args:
            sequence (:class:`qibolab.pulses.PulseSequence`): Pulse sequence to execute.
            options (:class:`qibolab.platforms.platform.ExecutionParameters`): Object holding the execution options.
            **kwargs: May need them for something

        Returns:
            Readout results acquired by after execution.
        """
        if options.nshots is None:
            options = replace(options, nshots=self.nshots)
```

Comment (DavidSarlle) [reply src/qibolab/platforms/multiqubit.py]: The point is that when we create the platform for qblox, we create Multiqubit platform, and the execute_pulse_sequence method executed is the one of the class Multiqubit, that is why we need to adde those lines also in Multiqubit.

**Comment by DavidSarlle** [comment]
This PR is ready to merge into main. Please @Jacfomg @andrea-pasquale @aorgazf

**Review by Jacfomg — APPROVED** [review]

**Review by andrea-pasquale — COMMENTED** [review]
Thanks for implementing this @DavidSarlle.
Please find below a few comments.
I still need to test in on hardware, as soon as I do it I will approve this PR.

**Comment by DavidSarlle** [comment]
> @DavidSarlle which runcard are you using? I tried with the one in the runcards repo but it does't work.

What is the error that you are getting?

**Comment by DavidSarlle** [comment]
> > @DavidSarlle which runcard are you using? I tried with the one in the runcards repo but it does't work.
> 
> What is the error that you are getting? I tried with a custom one.. Let me try with the one tht is in the repo

I just tried with the one that is in the repo and it is not working (the one that I tried did not contain the 2 qubit gates settings). It is raising the error:
  File "/nfs/users/david.fuentes/qibolab/src/qibolab/platforms/multiqubit.py", line 55, in reload_settings
    super().reload_settings()
  File "/nfs/users/david.fuentes/qibolab/src/qibolab/platform.py", line 114, in reload_settings
    self.pairs[pair].native_gates = TwoQubitNatives.from_dict(self.qubits, gatedict)
KeyError: (2, 3)

But I have not modified that part in the Platform file. Is it working for the rest of the platforms?

**Comment by andrea-pasquale** [comment]
> > > @DavidSarlle which runcard are you using? I tried with the one in the runcards repo but it does't work.
> > 
> > 
> > What is the error that you are getting? I tried with a custom one.. Let me try with the one tht is in the repo
> 
> I just tried with the one that is in the repo and it is not working (the one that I tried did not contain the 2 qubit gates settings). It is raising the error: File "/nfs/users/david.fuentes/qibolab/src/qibolab/platforms/multiqubit.py", line 55, in reload_settings super().reload_settings() File "/nfs/users/david.fuentes/qibolab/src/qibolab/platform.py", line 114, in reload_settings self.pairs[pair].native_gates = TwoQubitNatives.from_dict(self.qubits, gatedict) KeyError: (2, 3)
> 
> But I have not modified that part in the Platform file. Is it working for the rest of the platforms?

I have tested `tii1q_b1`. But I think the other platforms should be working, right @Jacfomg @rodolfocarobene @stavros11 ?

**Comment by DavidSarlle** [comment]
> topology: [[0, 2], [1, 2], [2, 3], [2, 4]]

please @andrea-pasquale check with the runcard located here: [david/fiw_qw5q_topology](https://github.com/qiboteam/qibolab_platforms_qrc/pull/11)

File: src/qibolab/instruments/qblox.py [hunk 5]
```diff
@@ -863,9 +863,10 @@ def process_pulse_sequence(self, instrument_pulses: PulseSequence, nshots: int,
                     body += initial_wait_instruction
 
                     for n in range(pulses.count):
-                        if (self.ports["i1"].hardware_demod_en or self.ports["o1"].hardware_mod_en) and pulses[
-                            n
-                        ].relative_phase != 0:
+                        # if (self.ports["i1"].hardware_demod_en or self.ports["o1"].hardware_mod_en) and pulses[
+                        #     n
+                        # ].relative_phase != 0:
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibolab/instruments/qblox.py:868]: Done @andrea-pasquale

File: src/qibolab/platforms/multiqubit.py [hunk 6]
```diff
@@ -4,11 +4,19 @@
 import yaml
 from qibo.config import log, raise_error
 
+from qibolab import AcquisitionType, AveragingMode, ExecutionParameters
```
[the remark this answers is not in the document]
Comment (DavidSarlle) [reply src/qibolab/platforms/multiqubit.py:7]: Done @andrea-pasquale

**Review by andrea-pasquale — APPROVED** [review]
Thanks @DavidSarlle
